cps/main.c

```diff
@@ -460,7 +461,9 @@ process_egress(struct cps_config *cps_conf, struct gatekeeper_if *iface,
 			}
 
 			memmove(new_eth_hdr, eth_hdr, sizeof(*new_eth_hdr));
-			fill_vlan_hdr(new_eth_hdr, iface->vlan_tag_be,
+			vlan_tag_be = eth_hdr->ether_type == RTE_ETHER_TYPE_IPV4 ?
+				iface->ipv4_vlan_tag_be : iface->ipv6_vlan_tag_be;
+			fill_vlan_hdr(new_eth_hdr, vlan_tag_be,
 				rte_be_to_cpu_16(eth_hdr->ether_type));
```

Comment:
`eth_hdr->ether_type` is big endian, while `RTE_ETHER_TYPE_IPV4` is host endian.